alf/utils/common.py

```diff
@@ -48,6 +48,13 @@
 from . import dist_utils, gin_utils
 
 
+def cuda_is_available():
+    # When CUDA_VISIBLE_DEVICES= (empty), simply skip the check.
+    # This is useful during hardware issues when the check hangs forever.
+    cuda_dev = os.getenv("CUDA_VISIBLE_DEVICES", default=None)
+    return (cuda_dev is None or cuda_dev != "") and torch.cuda.is_available()
```

Comment:
If CUDA_VISIBLE_DEVICES=="", torch.cuda.is_availble() will return False.

Reply:
When hardware works properly then yes, but if there is a hardware or driver problem, then this call could hang.  That's the reason for this change.